src/map/script.c

```diff
@@ -23204,6 +23229,74 @@ BUILDIN(navigateto)
 }
 
 /**
+ * Clan System: Add a player to a clan
+ */
+BUILDIN(clan_join)
+{
+	struct map_session_data *sd = NULL;
+	int clan_id = script_getnum(st, 2);
+
+	if (script_hasdata(st, 3))
+		sd = map->id2sd(script_getnum(st, 3));
+	else
+		sd = map->id2sd(st->rid);
+
+	if (sd == NULL) {
+		script_pushint(st, false);
+		return false;
+	}
+
+	if (clan->join(sd, clan_id))
+		script_pushint(st, true);
+	else
+		script_pushint(st, false);
+
+	return true;
+}
+
+/**
+ * Clan System: Remove a player from clan
+ */
+BUILDIN(clan_leave)
+{
+	struct map_session_data *sd = NULL;
+
+	if (script_hasdata(st, 2))
+		sd = map->id2sd(script_getnum(st, 2));
+	else
+		sd = map->id2sd(st->rid);
+
+	if (sd == NULL) {
+		script_pushint(st, false);
+		return false;
+	}
+
+	if (clan->leave(sd, false))
+		script_pushint(st, true);
+	else
+		script_pushint(st, false);
+
+	return true;
+}
+
+/**
+ * Clan System: Show clan emblem next to npc name
+ */
+BUILDIN(clan_master)
+{
+	struct block_list *bl = map->id2bl(st->oid);
+	int clan_id = script_getnum(st, 2);
```

Comment:
Prone to `<= 0` clan IDs as there are no checks or assertions below. This should ideally push `0` on failure (with errors messages), and `1` on success.